include/macroni/ASTMatchers/Kernel/RCUMatchers.hpp

```diff
@@ -0,0 +1,148 @@
+                 clang::ast_matchers::hasDescendant(
+                     clang::ast_matchers::parenExpr(clang::ast_matchers::has(
+                         clang::ast_matchers::parenExpr(
+                             clang::ast_matchers::has(
+                                 clang::ast_matchers::expr().bind("p"))))))))
+      .bind(name);
+}
+
+static const clang::ast_matchers::StatementMatcher rcu_deference_matcher =
+    make_rcu_dereference_matcher("");
+
+static const clang::ast_matchers::StatementMatcher rcu_deference_bh_matcher =
+    make_rcu_dereference_matcher("_bh");
+
+static const clang::ast_matchers::StatementMatcher rcu_deference_sched_matcher =
+    make_rcu_dereference_matcher("_sched");
+
+static const clang::ast_matchers::StatementMatcher rcu_assign_pointer_matcher =
+    clang::ast_matchers::doStmt(
+        clang::ast_matchers::allOf(
+            clang::ast_matchers::isExpandedFromMacro("rcu_assign_pointer"),
+            clang::ast_matchers::hasDescendant(
+                clang::ast_matchers::callExpr(clang::ast_matchers::allOf(
+                    clang::ast_matchers::callee(
+                        clang::ast_matchers::functionDecl(
+                            clang::ast_matchers::hasName("__write_once_size"))),
+                    clang::ast_matchers::hasArgument(
+                        0, clang::ast_matchers::hasDescendant(
+                               clang::ast_matchers::parenExpr(
+                                   clang::ast_matchers::has(
+                                       clang::ast_matchers::parenExpr(
+                                           clang::ast_matchers::has(
+                                               clang::ast_matchers::expr().bind(
+                                                   "p")))))))))),
+            clang::ast_matchers::hasDescendant(
+                clang::ast_matchers::callExpr(clang::ast_matchers::allOf(
+                    clang::ast_matchers::callee(
+                        clang::ast_matchers::functionDecl(
+                            clang::ast_matchers::hasName(
+                                "__builtin_constant_p"))),
+                    clang::ast_matchers::hasArgument(
+                        0, clang::ast_matchers::expr().bind("v")))))
+
+                ))
+        .bind("rcu_assign_pointer");
+
+static const clang::ast_matchers::StatementMatcher rcu_access_pointer_matcher =
+    clang::ast_matchers::stmtExpr(
+        clang::ast_matchers::allOf(
+            clang::ast_matchers::isExpandedFromMacro("rcu_access_pointer"),
+            clang::ast_matchers::hasDescendant(
+                clang::ast_matchers::parenExpr(clang::ast_matchers::has(
+                    clang::ast_matchers::parenExpr(clang::ast_matchers::has(
+                        clang::ast_matchers::expr().bind("p"))))))))
+        .bind("rcu_access_pointer");
+
+static const clang::ast_matchers::StatementMatcher rcu_replace_pointer_matcher =
```

Comment:
Why are all of these spilled in the global namespace compared to just a static members of the class that uses them? Are they used at more places?

Reply:
I just didn't think to make them static members; that's a great idea.
Can I use a `using namespace` inside the class definition then?